flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-postgres/src/main/java/org/apache/flink/cdc/connectors/postgres/factory/PostgresDataSourceFactory.java

```
@@ -369,23 +369,21 @@ private Optional<String> getValidateDatabaseName(String tables) {
                     String.format(
                             "Tables format must db.schema.table, can not 'tables' = %s",
                             TABLES.key()));
-            if (tableNameParts.length == 3) {
-                String currentDbName = tableNameParts[0];
+            String currentDbName = tableNameParts[0];
 
+            checkState(
+                    isValidPostgresDbName(currentDbName),
+                    String.format(
+                            "The value of option %s does not conform to PostgresSQL database name naming conventions",
+                            TABLES.key()));
+            if (dbName == null) {
+                dbName = currentDbName;
+            } else {
                 checkState(
-                        isValidPostgresDbName(currentDbName),
+                        dbName.equals(currentDbName),
```

postgres pipeline can only support ingest of tables under the same dbName, and should raise an exception if there are duplicate databases. checkState false throws an exception, duplicates the database !dbName=currentDBName.

---

I use the following configuration:
```
tables: aia_test.public.aia_t_icc_jjdb,aia_test.public.aia_t_icc_jjdb_\d{6},aia_test.public.aia_t_icc_jjdb_extend,aia_test.public.aia_t_icc_jjdb_extend_\d{6},aia_test.public.aia_t_vcs_fkdb,aia_test.public.aia_t_vcs_fkdb_\d{6}
```
Tip java.lang.IllegalStateException: The value of option tables all table names must have the same database name

Yes,  I'm going to write a unit test, so I'll have to wait a moment.

---

> Can you write a ut cover this?
Unit tests have been submitted